src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+      1 => 'Generic / Universal',
+      2 => 'Agriculture & Horticulture',
+      3 => 'Architecture & Construction',
+      4 => 'Arts & Culture',
+      5 => 'Automotive & Transport',
+      6 => 'Banking & Finance',
+      7 => 'Business & Commerce',
+      8 => 'Communication & Media',
+      9 => 'Compliance',
+      10 => 'Computer Hardware & Telecommunications',
+      11 => 'Computer Software & Networking',
+      12 => 'Electrical Engineering / Electronics',
+      13 => 'Energy & Environment',
+      14 => 'Food & Drink',
+      15 => 'General Healthcare',
+      16 => 'Law & Legal',
+      17 => 'Manufacturing / Industry',
+      18 => 'Marketing, Advertising & Fashion',
+      19 => 'Mechanical Engineering / Machinery',
+      20 => 'Military & Defence',
+      21 => 'Pharmaceutical & Clinical trials',
+      22 => 'Science & Chemicals',
+      23 => 'Specialist Healthcare (Machinery)',
+      24 => 'Specialist Healthcare (Practise)',
+      25 => 'Sports, Entertainment & Gaming',
+      26 => 'Travel Hospitality & Tourism',
+      27 => 'UK Gov (EU based)',
+      28 => 'UK Government',
+      29 => 'Veterinary Sciences',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function checkAvailable(TranslatorInterface $translator) {
+    if ($translator->getSetting('client_contact_key')) {
+      return AvailableResult::yes();
+    }
+    return AvailableResult::no(t('@translator is not available. Make sure it is properly <a href=:configured>configured</a>.', [
+      '@translator' => $translator->label(),
+      ':configured' => $translator->url(),
+    ]));
+  }
+
+  /**
+   * Creates new translation project at Thebigword.
+   *
+   * @param int $purchase_order_number
+   *   Translation job item id that is equals to the purchase order number.
+   * @param int $project_reference
+   *   Translation job item id that is equals to the project reference.
+   * @param string $required_by
+   *   Required date.
+   * @param string $quote_required
+   *   Is the quote required?
+   * @param string $category
+   *   The category of the translation.
+   * @param bool $review
+   *   True is Localize and review, False just Localize.
+   *
+   * @return int
+   *   Thebigword project id.
+   */
+  public function newTranslationProject($purchase_order_number, $project_reference, $required_by, $quote_required, $category, $review = TRUE) {
+    /** @var \DateTime $required_by */
+    $required_by->setTimezone(new \DateTimeZone('UTC'));
+    $datetime = $required_by->format('Y-m-d\TH:i:s');
+    $url = new Url('tmgmt_thebigword.callback');
+    $params = [
+      'PurchaseOrderNumber' => $purchase_order_number,
+      'ProjectReference' => $project_reference,
+      'RequiredByDateUtc' => $datetime,
+      'QuoteRequired' => $quote_required ? 'true' : 'false',
+      'SpecialismId' => $category,
+      'ProjectMetadata' => [
+        ['MetadataKey' => 'CMS User Name', 'MetadataValue' => \Drupal::currentUser()->getDisplayName()],
```

Comment:
Created this issue: [https://www.drupal.org/node/2675662](https://www.drupal.org/node/2675662)